### PR TITLE
chore: update samples to Akka 23.05

### PR DESCRIPTION
**Pull request by pvlugter**
Remove Lagom samples, and sync with the Akka Platform Guide samples on Akka 23.05 (before updating to 23.10). Uses Cinnamon 2.17.3 (from the akka dependencies BOM) or explicitly for sbt.

### CODE REVIEW

**Comment by pvlugter** [comment]
Also updated the secrets for the PR validation action, for commercial repository, as seemed to be invalid.

File: akka/shopping-cart-java/docker-compose.yml [hunk 1]
```diff
@@ -33,8 +33,6 @@ services:
     environment:
       KAFKA_ADVERTISED_HOST_NAME: localhost
       KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
-    volumes:
-      - /var/run/docker.sock:/var/run/docker.sock
```
Comment (sebastian-alfers) [inline akka/shopping-cart-java/docker-compose.yml:37]: Just curious: why was that removed?

File: akka/shopping-cart-java/src/main/java/shopping/cart/Main.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public class Main {
 
   // tag::SendOrderProjection[]
   public static void main(String[] args) {
-    ActorSystem<Void> system = ActorSystem.create(Behaviors.empty(), "ShoppingCartService");
+    ActorSystem<Void> system = ActorSystem.create(Behaviors.empty(), "shopping-cart-service");
```
Comment (sebastian-alfers) [inline akka/shopping-cart-java/src/main/java/shopping/cart/Main.java:29]: Was this a requirement, or more a convention?

File: akka/shopping-cart-scala/build.sbt [hunk 3]
```diff
@@ -2,9 +2,7 @@ name := "shopping-cart-service"
 
 organization := "com.lightbend.akka.samples"
 organizationHomepage := Some(url("https://akka.io"))
-licenses := Seq(
-  ("CC0", url("https://creativecommons.org/publicdomain/zero/1.0"))
-)
+licenses := Seq(("CC0", url("https://creativecommons.org/publicdomain/zero/1.0")))
```
Comment (sebastian-alfers) [inline akka/shopping-cart-scala/build.sbt:5]: I am not really into this, but are we good regarding our change to BSL?

File: akka/shopping-cart-scala/kubernetes/shopping-cart-service-cr.yml [hunk 4]
```diff
@@ -1,25 +0,0 @@
-apiVersion: "v1"
-kind: "Namespace"
-metadata:
-  name: "shopping"
----
-apiVersion: akka.lightbend.com/v1
-kind: AkkaMicroservice
```
Comment (sebastian-alfers) [inline akka/shopping-cart-scala/kubernetes/shopping-cart-service-cr.yml:7]: What is / was that? Our own custom resource or so? And we do not use / maintain it anymore?

**Review by sebastian-alfers — APPROVED** [review]
LGTM

File: akka/shopping-cart-java/docker-compose.yml (hunk 1 above)
Comment (pvlugter) [reply akka/shopping-cart-java/docker-compose.yml:37]: The samples here are synced from the Akka Guide samples. You can check the history there:

https://github.com/akka/akka-guide

Comment (pvlugter) [reply akka/shopping-cart-java/docker-compose.yml:37]: For example: https://github.com/akka/akka-guide/pull/711

File: akka/shopping-cart-java/src/main/java/shopping/cart/Main.java (hunk 2 above)
Comment (pvlugter) [reply akka/shopping-cart-java/src/main/java/shopping/cart/Main.java:29]: Check the Akka Guide for changes. From https://github.com/akka/akka-guide/pull/832, so could be to simplify bootstrap in Kubernetes? (by having the service and actor system names the same)

File: akka/shopping-cart-scala/build.sbt (hunk 3 above)
Comment (pvlugter) [reply akka/shopping-cart-scala/build.sbt:5]: Assume so, and that we're using CC0 across the Akka samples.
